test/unit/test_ArithLogicApi.cc

```diff
@@ -0,0 +1,46 @@
+//
+// Created by Antti on 16.09.21.
+//
+
+#include <gtest/gtest.h>
+#include <ArithLogic.h>
+#include <StringConv.h>
+
+class ArithLogicApiTest: public ::testing::Test {
+public:
+    ArithLogicApiTest()
+        : lraLogic(ArithLogic::ArithType::LRA)
+        , liaLogic(ArithLogic::ArithType::LIA)
+        , liraLogic(ArithLogic::ArithType::LIRA)
+    {}
+    ArithLogic lraLogic;
+    ArithLogic liaLogic;
+    ArithLogic liraLogic;
+};
+
+TEST_F(ArithLogicApiTest, test_LRA) {
+    PTRef c1 = lraLogic.mkConst("213");
+    PTRef c2 = lraLogic.mkRealVar("a");
+    ASSERT_NO_THROW(lraLogic.mkPlus(c1, c2));
+    ASSERT_THROW(lraLogic.mkTimes(c2, c2), LANonLinearException);
+    ASSERT_THROW(lraLogic.mkIntVar("x"), OsmtApiException);
+    ASSERT_THROW(lraLogic.mkIntConst(2), OsmtApiException);
+}
+
+TEST_F(ArithLogicApiTest, test_LIA) {
+    PTRef c1 = liaLogic.mkConst("213");
+    PTRef c2 = liaLogic.mkIntVar("a");
+    ASSERT_NO_THROW(liaLogic.mkPlus(c1, c2));
+    ASSERT_THROW(liaLogic.mkTimes(c2, c2), LANonLinearException);
+    ASSERT_THROW(liaLogic.mkRealVar("a"), OsmtApiException);
+    ASSERT_THROW(liaLogic.mkRealConst(2), OsmtApiException);
+}
+
+TEST_F(ArithLogicApiTest, test_Mixed) {
+    PTRef c1 = liraLogic.mkConst("213");
```

Comment:
This I was not quite sure about.
So `ArithLogic::mkConst` when parsing numbers like this (`213`) will create `Real` constant if it does not support integers and `Integer` constant otherwise?

Reply:
Yes, it should work like this.  I'll add some tests that make this more explicit (and which identified a problem in the code, so I'll add a fix as well).